### PR TITLE
doc(rest): add rest-options group

### PR DESCRIPTION
**Pull request by scotthart**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10249)

### CODE REVIEW

File: google/cloud/doc/options.dox [hunk 1]
```diff
@@ -85,4 +85,10 @@ ignored by other libraries). In some cases, the option may be specific to a
 Other services in the same library (e.g., `kms::EkmServiceClient`) will ignore
 this option.  Likewise, other services in different libraries (e.g.
 `pubsub::Publisher`) will also ignore the option.
+
+@defgroup rest-options Options only applicable when using REST transport.
+@ingroup options
+
+These Options are used with libraries that either only support REST transport,
+or with libraries that the user has elected to enable gRPC to HTTP transcoding.
```
Comment (devbww) [inline google/cloud/doc/options.dox:93]: s/that/where/ perhaps?

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10249)* status: 0 of 2 files reviewed, 1 unresolved discussion (waiting on @devbww)

___
*[`google/cloud/doc/options.dox` line 93 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10249#-NGscZdDzryjLu1WoHkE-r1-93:-NGwJByC8bB9t8Mjl3Z2:b-xrcooz) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/76112fd3fa57cdf6970f27bbeb73d70f80615610/google/cloud/doc/options.dox#L93)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

s/that/where/ perhaps?
</blockquote></details>

opted for "which"

**Review by devbww — COMMENTED** [review]
> Reviewed 2 of 2 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10249)* status: all files reviewed, 1 unresolved discussion (waiting on @scotthart)

___
*[`google/cloud/doc/options.dox` line 93 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10249#-NGscZdDzryjLu1WoHkE-r1-93:-NGwNu1z3pLxqa92vpyH:b-xlbsci) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/76112fd3fa57cdf6970f27bbeb73d70f80615610/google/cloud/doc/options.dox#L93)):*
<details><summary><i>Previously, scotthart (Scott Hart) wrote…</i></summary><blockquote>

opted for "which"
</blockquote></details>

I think the "that" that you changed should stay a "that" ... it is a defining clause.

But the "that" I was referring to is in the next line ... "with libraries where the user ...".

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10249)* status: 1 of 2 files reviewed, all discussions resolved (waiting on @devbww)

___
*[`google/cloud/doc/options.dox` line 93 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10249#-NGscZdDzryjLu1WoHkE-r1-93:-NGwkPuTBEGesUK1RmHs:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/76112fd3fa57cdf6970f27bbeb73d70f80615610/google/cloud/doc/options.dox#L93)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

I think the "that" that you changed should stay a "that" ... it is a defining clause.

But the "that" I was referring to is in the next line ... "with libraries where the user ...".
</blockquote></details>

Done.

File: google/cloud/doc/options.dox [hunk 2]
```diff
@@ -85,4 +85,10 @@ ignored by other libraries). In some cases, the option may be specific to a
 Other services in the same library (e.g., `kms::EkmServiceClient`) will ignore
 this option.  Likewise, other services in different libraries (e.g.
 `pubsub::Publisher`) will also ignore the option.
+
+@defgroup rest-options Options only applicable when using REST transport.
```
Comment (devbww) [inline google/cloud/doc/options.dox:89]: It looks like there probably shouldn't be a trailing period (on a heading).

**Review by devbww — APPROVED** [review]

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10249)* status: 1 of 2 files reviewed, 1 unresolved discussion (waiting on @devbww)

___
*[`google/cloud/doc/options.dox` line 89 at r3](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10249#-NGscZdDzryjLu1WoHkE-r3-89:-NGx-bHp1xD0xCUJv5iG:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/ad97ce062cdab9d9f0950dd9fb539333dca55ec7/google/cloud/doc/options.dox#L89)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

It looks like there probably shouldn't be a trailing period (on a heading).
</blockquote></details>

Done.

**Pull request merged**